descartes_moveit/src/ikfast_moveit_state_adapter.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Jonathan Meyer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "descartes_moveit/ikfast_moveit_state_adapter.h"
+
+#include <cassert>
+#include <eigen_conversions/eigen_msg.h>
+#include <ros/node_handle.h>
+
+const static std::string default_base_frame = "base_link";
+const static std::string default_tool_frame = "tool0";
+
+// Compute the 'joint distance' between two poses
```

Comment:
L2 is probably mathematically more correct.  However, an L1 is more efficient (no square root or multiplication)  If all that is considered from the distance function is a value comparison (> or <) and not a true distance difference, then the resulting comparison is the same whether an L2 or L1 norm is used.